crates/pathfinder/src/state/sync.rs

```diff
@@ -349,7 +349,7 @@ where
                         let tx = db_conn
                             .transaction_with_behavior(TransactionBehavior::Immediate)
                             .context("Create database transaction")?;
-                        let new_root = update_starknet_state(&tx, &state_update).context("Updating Starknet state")?;
+                        let new_root = update_starknet_state(&tx, &state_update).context("Updating Starknet state")?.into();
```

Comment:
I think I would expect this function itself to already calculate the state commitment; not a big fan of it hiding in the `.into` here.

Ah I see its a result of the storage decision which means you need the separate values here still.. hmm.